### PR TITLE
SLCORE-630 Synchronize the storage when credentials are updated

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java [hunk 1]
```diff
@@ -224,6 +226,21 @@ public void onBindingChanged(BindingConfigChangedEvent event) {
...
+  @EventListener
+  public void onConnectionCredentialsChanged(ConnectionCredentialsChangedEvent event) {
+    if (!fullSynchronizationEnabled) {
+      return;
+    }
+    var connectionId = event.getConnectionId();
+    LOG.debug("Synchronizing connection '{}' after credential changed", connectionId);
+    var bindingsForUpdatedConnection = configurationRepository.getBoundScopesByConnection(connectionId)
+      .stream()
+      .map(b -> new BoundScope(connectionId, connectionId, b.getSonarProjectKey()))
```
Comment (damien-urruty-sonarsource) [inline core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java:238]: This line looks suspicious, the first expected parameter is the config scope ID. Do you need to map at all? The `getBoundScopesByConnection` should return what you need

File: core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java [hunk 2]
```diff
@@ -224,6 +226,21 @@ public void onBindingChanged(BindingConfigChangedEvent event) {
     }
   }
 
+  @EventListener
+  public void onConnectionCredentialsChanged(ConnectionCredentialsChangedEvent event) {
+    if (!fullSynchronizationEnabled) {
+      return;
+    }
+    var connectionId = event.getConnectionId();
+    LOG.debug("Synchronizing connection '{}' after credential changed", connectionId);
```
Comment (damien-urruty-sonarsource) [inline core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java:235]: Maybe it's valid but I often see `credentials` used with plural

**Review by damien-urruty-sonarsource — APPROVED** [review]
Small comments, else LGTM

File: core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java (hunk 1 above)
Comment (henryju) [reply core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java:238]: Good catch. At some point, there was a mix of BoundScope and BoundConfigScope, and I remember I was struggling to convert from one to another.
Seems simpler now (maybe after you merged your changes).

**Pull request merged**